### ISSUE TITLE
null pointer write doesn't break until verification stage

### ISSUE THREAD

**Issue by stskeeps** [risc0/risc0#625]
as discussed with jeremy, 

[2023-06-13T11:20:23Z TRACE risc0_zkvm::prove::exec] [138325] pc: 0x0701f5ac, insn: 0x00072583 => lw x11, 0(x14)
[2023-06-13T11:20:23Z TRACE risc0_zkvm::prove::exec]   imm=0x00000000, rs1=x14 -> 0x00000000, rs2=x0 -> 0x00000000, Addr = 0x00000000, lowBits = 0, loaded = 0x00000000
[2023-06-13T11:20:23Z TRACE risc0_zkvm::prove::exec]   fillByte = 0x00, extended: 0x00000000

[2023-06-13T11:23:45Z DEBUG risc0_zkp::verify::host] > compute_polynomial
[2023-06-13T11:23:45Z DEBUG risc0_zkp::verify::host] < compute_polynomial
thread 'main' panicked at 'called Result::unwrap() on an Err value: Verification indicates proof is invalid', host/src/main.rs:36:35

"Oh, interesting.  The zero page is a fault to read or write (purposely), but I wonder if the witness generations isn’t tightly constrained enough on that, but the actual proof polynomial is.  That would explain the behavior.  Do you have a repo you could send?"

repo here: https://github.com/stskeeps/nullaccess-bug

uarch_combined.o is rv32im compiled object made from the corresponding c file w/o optimization, forcing a null pointer write

**Comment by flaub** [comment]
@SchmErik This might be related to #672 

Generally we need to improve the executor esp. regarding memory accesses.

**Comment by SchmErik** [comment]
> @SchmErik This might be related to #672

Yeah, I think this is related but I think this is also a separate issue in that the prover allows a null pointer to be written.
> 
> Generally we need to improve the executor esp. regarding memory accesses.

Fixing this in the executor means that it won't pass through the prover if you're using the executor and prover like we suggest in the starter template but I think we may want to look at patching the prover too

**Comment by flaub** [comment]
The prover currently should prevent null pointer writes, because this is enforced by circuit constraints. It's the executor that's currently allowing it I think.

**Comment by flaub** [comment]
@jbruestle Is there potentially a bug in the prover here?

**Comment by SchmErik** [comment]
Removing the fixes tag from #727 for now
